src/storage/assets/assetsActions.ts

```diff
@@ -41,3 +45,43 @@ export const syncNetworkTokensInfo = createAsyncThunk('assets/syncNetworkTokensI
 
   return metadata
 })
+
+export const syncUnknownTokensInfo = createAsyncThunk(
+  'assets/syncUnknownTokensInfo',
+  async (unknownTokenIds: Asset['id'][]): Promise<SyncUnknownTokensInfoResult> => {
+    const results = {
+      tokens: [],
+      nfts: []
+    } as SyncUnknownTokensInfoResult
+
+    for await (const id of unknownTokenIds) {
+      const type = await client.node.guessStdTokenType(id)
+
+      try {
+        if (type === 'fungible' || type === undefined) {
+          const tokenMetadata = await client.node.fetchFungibleTokenMetaData(id)
+
+          results.tokens.push({
+            id,
+            ...omit(tokenMetadata, ['totalSupply'])
+          })
+        } else if (type === 'non-fungible') {
+          const { tokenUri, collectionAddress } = await client.node.fetchNFTMetaData(id)
+          const nftData = await exponentialBackoffFetchRetry(tokenUri).then((res) => res.json())
```

Comment:
The type of `nftData` is `any` since we can't know what the `tokenUri` will return.

Reply:
Yes. We need to investigate if the team pushed for a standardisation (I know that Cheng wanted to). On the explorer, I consider the current [format](https://github.com/alephium/desktop-wallet/pull/725#discussion_r1244780430).

Reply:
Hongchao just confirmed that they encourage using this format, it's here: https://github.com/alephium/alephium-web3/blob/master/packages/web3/std/nft_interface.ral

Reply:
Gotcha! I would still make the type properties optional so that we don't get any white screens in case of a rogue NFT.